front/app/components/Form/Components/Layouts/utils.ts

```diff
@@ -66,7 +66,7 @@ export const getFilteredDataForUserPath = (
   userRoute: PageType[],
   data: any
 ) => {
-  const filteredData = { data };
+  const filteredData = { data: { ...data } };
```

Comment:
This was modifying the passed `data`, so the function was not pure. By cloning the data like this it is.